internal/getproviders/package_authentication.go

```diff
@@ -3,11 +3,60 @@ package getproviders
 import (
 	"bytes"
 	"crypto/sha256"
+	"encoding/hex"
 	"fmt"
 	"io"
+	"log"
 	"os"
+	"strings"
+
+	"golang.org/x/crypto/openpgp"
+	openpgpArmor "golang.org/x/crypto/openpgp/armor"
+	openpgpErrors "golang.org/x/crypto/openpgp/errors"
+)
+
+type packageAuthenticationResult int
+
+const (
+	verifiedChecksum packageAuthenticationResult = iota
+	hashicorpProvider
+	partnerProvider
+	communityProvider
 )
 
+// PackageAuthenticationResult is returned from a PackageAuthentication
+// implementation. It is a mostly-opaque type intended for use in UI, which
+// implements Stringer and includes an optional Warning field.
+//
+// A failed PackageAuthentication attempt will return an "unauthenticated"
+// result, which is represented by nil.
+type PackageAuthenticationResult struct {
+	result  packageAuthenticationResult
+	Warning string
+}
+
+func (t *PackageAuthenticationResult) String() string {
+	if t == nil {
+		return "unauthenticated"
+	}
+	return []string{
+		"verified checksum",
+		"HashiCorp provider",
+		"Partner provider",
+		"community provider",
```

Comment:
The uppercase "P" on "Partner" with the lowercase "c" on "community" feels a little "off" here. Was that intentional? :thinking: (I think it might be an accidental leftover from where it previously said "HashiCorp Partner"?)

Reply:
This is definitely a mistake. Based on [the Registry docs](https://github.com/hashicorp/terraform/blob/3856dacc1ef7c1bb7c182edac5411e2e8a25613e/website/docs/registry/providers/tiers.html.md), I think "HashiCorp provider" should also be "official provider". Will fix!